module/zfs/arc.c

```diff
@@ -4066,10 +4121,14 @@ arc_evict_state(arc_state_t *state, arc_buf_contents_t type, uint64_t spa,
 {
 	uint64_t total_evicted = 0;
 	multilist_t *ml = &state->arcs_list[type];
-	int num_sublists;
 	arc_buf_hdr_t **markers;
+	unsigned num_sublists = multilist_get_num_sublists(ml);
+	evict_arg_t *evarg = NULL;
+	uint_t nthreads = zfs_arc_evict_threads == 1 ?
+	    zfs_arc_evict_threads_max :
+	    MIN(zfs_arc_evict_threads, zfs_arc_evict_threads_max);
```

Comment:
We could limit `nthreads` to `num_sublists`.  Multiple threads per sublist might not have much sense.  It should normally be true, but `zfs_multilist_num_sublists` if configurable.

Reply:
I think it is worth checking num_sublist when zfs_arc_evict_threads_max is computed. Updated.

Reply:
While it may be some better in practice by not creating extra threads, it dives into internal multilist details, which is not good.  I wont insist, but I would not do it this way myself.